packages/airnode-node/src/handlers/process-http-signed-data-request.ts

```diff
@@ -17,29 +19,43 @@ export async function processHttpSignedDataRequest(
 
   const endpoints = find(config.ois, ['title', trigger.oisTitle])?.endpoints;
   const endpoint = find(endpoints, ['name', trigger.endpointName]);
+  const decodedParameters = evm.encoding.safeDecode(parameters);
+
   if (!endpoint) {
     return [new Error(`No endpoint definition for endpoint ID '${endpointId}'`), null];
   }
 
   // TODO: There should be an TS interface for required params
-  if (!parameters._templateId) {
-    return [
-      new Error(`You must specify "_templateId" for the requestId/subscriptionId in the request parameters.`),
-      null,
-    ];
+  if (!decodedParameters) {
+    return [new Error(`Request contains invalid encodedParameters: ${parameters}`), null];
   }
 
   const requestId = randomHexString(16);
   const logOptions = buildBaseOptions(config, { requestId });
   const airnodeAddress = wallet.getAirnodeWallet(config).address;
+
+  const template = <ApiCallTemplate>{
```

Comment:
I haven't seen this syntax used before (?)

```suggestion
  const template: ApiCallTemplate = {
```

Reply:
so the `getExpectedTemplateId()`  expects a `ApiCallTemplate ` type . The problem being this type needs an `id` which we are trying to generate. I wasn't sure if I should be creating a new type so I just type casted the existing ApiCallTemplate without the `id` . this is functionally identical to:

```
  const template = {
    airnodeAddress,
    endpointId,
    encodedParameters: parameters,
  } as ApiCallTemplate;
  ```
  
  Maybe I should create a new type without id ?

Reply:
I'm not sure I like this approach... So this would loses any type safety we get from the type? As a general rule, I think if a function expects a certain type, then we shouldn't try to force our way around that. Would changing the input type on `getExpectedTemplateId` be a lot of work?

@amarthadan @Siegrift what do you guys think?

Reply:
I agree with @andreogle. If you would use
```ts
type ApiCallTemplateForValidation = Omit<ApiCallTemplate, 'id'>
```
as an argument type for `getTemplateIdValidationFields` and `getExpectedTemplateId` it should work without any other changes.

Feel free to come up with a better name for that type :sweat_smile: